File: 1260.py
import sys
sys.setrecursionlimit(10**6)

n, m, v = map(int,input().split())


node = []
for j in range(n):
    node.append([])

for i in range(m):
    a, b = map(int,input().split())
    if b not in node[a-1]:
        node[a-1].append(b)
    for k in range(len(node[a-1])):
        if a not in node[node[a-1][k]-1]:
            node[node[a-1][k]-1].append(a)

for ll in range(len(node)):
    node[ll].sort()

visited_dfs = []
def dfs(first):
    visited_dfs.append(first)
    print(first, end=" ")
    for i in range(len(node[first-1])):
        if node[first-1][i] not in visited_dfs:
            dfs(node[first-1][i])

dfs(v)
print()
index = 0
visited_bfs = [v]
def bfs(first, index):
    for i in range(len(node[first-1])):
        if node[first-1][i] not in visited_bfs:
            visited_bfs.append(node[first-1][i])
    index += 1
    if index < len(visited_bfs):
        bfs(visited_bfs[index],index)

bfs(v,index)
for nn in range(len(visited_bfs)):
    print(visited_bfs[nn], end=" ")

print()

File: 1260.cc
// #include <bits/stdc++.h>
// using namespace std;

// int n,m,v;
// vector<vector<int>> arr;

// int main(){
//     ios::sync_with_stdio(0);
//     cin.tie(0);

//     cin >> n >> m >> v;

//     arr.resize(n+1);

//     for(int i = 0;i < m; i++){
//         int a, b;
//         cin >> a >> b;
//         arr[a].push_back(b);
//         arr[b].push_back(a);
//     }

//     int visit_[1001];

//     queue<int> dfs;
    




//     int visit[1001];

//     stack<int> bfs;
//     bfs.push(1);

//     while (!bfs.empty())
//     {
//         int pos = bfs.top();
//         visit[pos] = 1;
//         bfs.pop();


//         cout << pos << " ";

//         for(int i = 0; i < arr[pos].size(); i++){
//             if(visit[arr[pos][i]] == 1) continue;        
//             sort(arr[i].begin(), arr[i].end(), greater<int>());
//             bfs.push(arr[pos][i]);
//             visit[arr[pos][i]] = 1;
//         }
//     }
//     cout << "\n";


// }


#include <bits/stdc++.h>
using namespace std;

int arr[1001][1001];
bool visited_dfs[1001];
bool visited_bfs[1001];

int n, m, v;

void dfs(int index){
    visited_dfs[index] = true;
    cout << index << " ";

    for(int i = 1; i <= n; i++){
        if(arr[index][i] == 1 && !visited_dfs[i]){
            dfs(i);
        }
    }
}

void bfs(){
    queue<int> q;
    q.push(v);
    visited_bfs[v] = true;

    while(!q.empty()){
        int now = q.front();
        q.pop();

        cout << now << " ";

        for(int i = 1; i <= n; i++){
            if(arr[now][i] == 1 && !visited_bfs[i]){
                q.push(i);
                visited_bfs[i] = true;
            }
        }
    }

}


int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);

    cin >> n >> m >> v;

    for(int i = 0; i < m; i++){
        int a,b;
        cin >> a >> b;

        arr[a][b] = 1;
        arr[b][a] = 1;
    }

    dfs(v);
    cout << endl;
    bfs();
}